src/bots/liquidator.ts

```diff
@@ -412,6 +427,179 @@ export class LiquidatorBot implements Bot {
 		return healthy;
 	}
 
+	private async driftSpotTrade(
+		orderDirection: PositionDirection,
+		marketIndex: number,
+		standardizedTokenAmount: BN
+	) {
+		const start = Date.now();
+		this.driftClient
+			.placeSpotOrder(
+				getMarketOrderParams({
+					marketIndex: marketIndex,
+					direction: orderDirection,
+					baseAssetAmount: standardizedTokenAmount,
+					reduceOnly: true,
+				})
+			)
+			.then((tx) => {
+				logger.info(
+					`closing spot position for market ${marketIndex.toString()}: ${tx} `
+				);
+			})
+			.catch((e) => {
+				logger.error(
+					`Error trying to close spot position for market ${marketIndex}`
+				);
+				console.error(e);
+				webhookMessage(
+					`[${
+						this.name
+					}]: :x: error trying to close spot position on market ${marketIndex} \n:${
+						e.stack ? e.stack : e.message
+					} `
+				);
+			})
+			.finally(() => {
+				this.sdkCallDurationHistogram.record(Date.now() - start, {
+					method: 'placeSpotOrderDrift',
+				});
+			});
+	}
+
+	private async jupiterSpotSwap(
+		orderDirection: PositionDirection,
+		spotMarketIndex: number,
+		standardizedTokenAmount: BN,
+		route: Route
+	) {
+		let outMarketIndex: number;
+		let inMarketIndex: number;
+		let jupSwapMode: SwapMode;
+		if (isVariant(orderDirection, 'long')) {
+			// sell USDC, buy spotMarketIndex
+			inMarketIndex = 0;
+			outMarketIndex = spotMarketIndex;
+			jupSwapMode = 'ExactOut';
+		} else {
+			// sell spotMarketIndex, buy USDC
+			inMarketIndex = spotMarketIndex;
+			outMarketIndex = 0;
+			jupSwapMode = 'ExactIn';
+		}
+
+		const start = Date.now();
+		this.driftClient
+			.swap({
+				jupiterClient: this.jupiterClient,
+				outMarketIndex,
+				inMarketIndex,
+				amount: standardizedTokenAmount,
+				swapMode: jupSwapMode,
+				route,
+			})
+			.then((tx) => {
+				logger.info(
+					`closing spot position for market ${spotMarketIndex.toString()}: ${tx} `
+				);
+			})
+			.catch((e) => {
+				logger.error(
+					`Error trying to ${getVariant(
+						orderDirection
+					)} spot position for market ${spotMarketIndex} on jupiter`
+				);
+				console.error(e);
+				webhookMessage(
+					`[${this.name}]: :x: error trying to ${getVariant(
+						orderDirection
+					)} spot position on market on jupiter ${spotMarketIndex} \n:${
+						e.stack ? e.stack : e.message
+					} `
+				);
+			})
+			.finally(() => {
+				this.sdkCallDurationHistogram.record(Date.now() - start, {
+					method: 'placeAndTakeSpotOrderJupiter',
+				});
+			});
+	}
+
+	private async determineBestSpotSwapRoute(
+		spotMarketIndex: number,
+		orderDirection: PositionDirection,
+		baseAmountIn: BN
+	): Promise<Route | undefined> {
+		const oraclePriceData =
+			this.driftClient.getOracleDataForSpotMarket(spotMarketIndex);
+		const dlob = await this.userMap.getDLOB(oraclePriceData.slot);
+		if (!dlob) {
+			logger.error('afiled to load DLOB');
+			return undefined;
```

Comment:
If we cant load dlob, should we fallback to jup? Or at least see if we can get jup route?